tsdb/db.go

```diff
@@ -1393,3 +1435,88 @@ func exponential(d, min, max time.Duration) time.Duration {
 	}
 	return d
 }
+
+// ReadyStorage implements the Storage interface while allowing to set the actual
```

Comment:
Where is this used?

Reply:
https://github.com/prometheus/prometheus/blob/5efee4f980ea140a4a6972ab06da79060e4302ce/cmd/prometheus/main.go#L337

Reply:
This seems like something that'd better live in main. startTimeMargin also seems a bit dubious in pure DB terms.

Reply:
Cool, will look it through.

Reply:
done :+1: